superset/proxy/api.py

```diff
@@ -0,0 +1,190 @@
+    )
+    def get_userid(self, user_id: str, **_kwargs: Any) -> Response:
+        """
+        This is a function which will obtain an Alfred OBO token based on the
+        current logged in user, and will then send a request to Alfred to see
+        if the passed in user_id is in any reports/incidents
+        """
+        if (self.ALFRED_SCOPE is None or self.ALFRED_URL is None):
+            return self.error_obtaining_response("Alfred", "No Alfred Scope and/or No Alfred URL")
+
+        try:
+            alfred_token = security_manager.get_on_behalf_of_access_token_with_cache(current_user.username,
+                                                                                    self.ALFRED_SCOPE,
+                                                                                    'alfred',
+                                                                                    cache_result=True)
+            if not alfred_token:
+                raise Exception("Unable to fetch Alfred token")
+        except requests.exceptions.HTTPError as err:
+            return self.error_obtaining_token("Alfred", err)
+        except Exception as err:
+            return self.error_obtaining_token("Alfred", err)
+        else:
+            headers = CaseInsensitiveDict()
+            headers["Accept"] = "application/json"
+            headers["Authorization"] = f"Bearer { alfred_token }"
+            url = (
+                self.ALFRED_URL
+                + "/rest/search/cypher?expression=MATCH%20(email%3AEMAIL_ADDRESS)%20WHERE%20email.value%20IN%20%5B%22"
+                + user_id
+                + "%22%5D%20RETURN%20email.value%2C%20email.maliciousness%2C%20email.uri"
+            )
+            alfred_resp = ""
+
+            try:
+                alfred_resp = requests.get(url, headers=headers, verify=self.SSL_CERT)
+            except requests.exceptions.ConnectionError as err:
+                return self.error_obtaining_response("Alfred", err)
+
+            refresh_resp_json = json.loads(
+                alfred_resp.content.decode("utf8", "replace")
+            )
+            print(refresh_resp_json)
+            return self.attach_url(200, self.ALFRED_URL, False, refresh_resp_json)
+
+    @expose("/alfred/ip_string/<string:ip_string>", methods=["GET"])
+    @event_logger.log_this_with_context(
+        action=lambda self, *args, **_kwargs: f"{self.__class__.__name__}.get",
+        log_to_statsd=False,  # pylint: disable-arguments-renamed
+    )
+    def get_ipstring(self, ip_string: str, **_kwargs: Any) -> Response:
+        """
+        This is a function which will obtain an Alfred OBO token based on the
+        current logged in user, and will then send a request to Alfred to see
+        if the passed in ip_string is in any reports/incidents
+        """
+        if (self.ALFRED_SCOPE is None or self.ALFRED_URL is None):
+            return self.error_obtaining_response("Alfred", "No Alfred Scope and/or No Alfred URL")
```

Comment:
Same comment as above. Is it worth splitting these into two code paths?

Reply:
Fixed in the latest commit. Code was changed to be:

```suggestion
        if (self.ALFRED_SCOPE is None and self.ALFRED_URL is None):
            return self.error_obtaining_response("Alfred", "No Alfred Scope and No Alfred URL")
        if (self.ALFRED_SCOPE is None):
            return self.error_obtaining_response("Alfred", "No Alfred Scope")
        if (self.ALFRED_URL is None):
            return self.error_obtaining_response("Alfred", "No Alfred URL")
```